modules/core/include/opencv2/core/hal/intrin_avx.hpp

```diff
@@ -198,14 +198,12 @@ struct v_uint16x16
     __m256i val;
 
     explicit v_uint16x16(__m256i v) : val(v) {}
-    v_uint16x16(ushort v0,  ushort v1,  ushort v2,  ushort v3,
-                ushort v4,  ushort v5,  ushort v6,  ushort v7,
-                ushort v8,  ushort v9,  ushort v10, ushort v11,
-                ushort v12, ushort v13, ushort v14, ushort v15)
```

Comment:
This change is not acceptable.

You break already existed API.

---

Again, OpenCV SIMD UI is C++ based with helper custom register types.

Reply:
> This change is not acceptable.
> 
> You break already existed API.
> 

Hi @alalek  - yes, I understand that there are concerns in doing this. However, I think I have provided enough data with examples in 20562 and 20640 that shows that the change will not affect code generation.

> Again, OpenCV SIMD UI is C++ based with helper custom register types.

Yes, and that should be changed too. I think that we can make the C++ UI to use fixed size vectors and scalable size vectors depending on the target architecture. I totally get your concerns: it is a big change that seems to be too far away from the architectures that OpenCV supports.

Is there a way we could agree an exploratory development of a C++ vector API for scalable vectors? I'd like to propose we create a development branch where to do this. We will merge it into the main product only when we think we have reached a level of stability and performance that is good for the library and the developers using it. Do you think this would be a good approach? 

I am happy also to explore other proposals. I'd like to avoid the one you suggested to create a version of OpenCV that uses SVE for fixed lengths vectors. The cost of this approach is definitely lower on the short term, but it will be more expensive on the long term. We'll have to change the C++ UI anyway if it turns out we want to create a version of OpenCV that runs on every SVE implementation. Also, please notice that the size of the vector registers in SVE can also be set by the OS/hypervisor at program start: this means that in theory, for hardware that have an Nx128 bit SVE implementation, we could end up having to support all vector sizes from 128x1, 128x2, .... 128xN. 

Finally, I think that the concept of scalable vector is not as niche as it might look. There are several players other than Arm that are using scalable vectors in their architectures.

@alalek @asmorkalov @vpisarev  - I'd like to hear what you think on these topics and on the approach I am suggesting.

Kind regards,

Francesco